src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static int INITIAL_SIZE = 10;
+
+    private Object[] array = new Object[INITIAL_SIZE];
+    private int numberOfElements = 0;
+
     @Override
     public void add(T value) {
-
+        checkWhetherResize();
+        array[numberOfElements++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > numberOfElements) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        checkWhetherResize();
+        System.arraycopy(array, index, array, index + 1, numberOfElements - 1);
```

Comment:
probably we should shift the amount of elements from the right side of index?

Reply:
that's what we do. or I didn't get the question

Reply:
you are shifting all elements minus one

Reply:
changed